snapcore/core20
# spread: add spread testing for the core20 repository

Most of this is setup from how spread is setup in our core-initrd repo. This PR adds the following:

Spread tests:
 - Verifies boot
 - Verifies partition exists
 - Verifies that services are running
 - TPM/Secureboot

Spread backends:
 - GCE (for CI, a copy of the setup from core-initrd)
 - QEmu (for local testing)
 - LXD (unusable unless you use my fork of spread github.com/meulengracht/spread).

Documentation:
 - How to test using QEmu
 - How to test using LXD

So I'm creating this PR already now, I wanted to wait for my spread PR to land originally, but I can see that it might take a while, and we would really like to see spread testing a lot quicker than that. Fortunately we also support testing using QEmu backend, so people can use that locally for now, so just avoid ```spread lxd-nested``` unless you want to install the spread from my fork.

If we get this PR merged I'll create an identical PR for core22 repository. For now I'm starting with this so I don't have to make any changes twice.
